mozilla/balrog
# Bugfix 1365658 fix priority & background rate validators

This PR is to resolve [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1365658)

#### What does this PR do?
To remove the validation hack that had been implemented earlier and replace with front-end validation.
#### Description of Task to be completed?
1. Remove the existing hack.
2. Implement validation on the front end.
#### How should this be manually tested?
1. Once balrog is running on http://127.0.0.1:8080 navigate to `Rules` and click on `Add a New Rule`.
2. Test by inputing invalid data for `Rate`- either negative numbers(eg `-40`), over 100(eg 144) or non-digit characters(eg 'x', or '&').
3. Also input invalid data for `Priority` - either negative numbers or non-digit characters. 
4. A error label will appear below the respective input with a relevant error message when the user clicks on the save changes button.
 
#### What are the relevant bugzilla stories?
https://bugzilla.mozilla.org/show_bug.cgi?id=1365658
#### Screenshots (if appropriate)
<img width="1353" alt="screen shot 2017-09-29 at 09 16 38" src="https://user-images.githubusercontent.com/28535137/31016502-a3e678dc-a52d-11e7-947f-8b736e179cf8.png">

#### Questions:
1. I need feedback on the error message framing.
2. If I can also have all the other input fields highlighted by the alert class on validation failure. Currently, only a message is posted at the bottom of the page after dismissing the sweetAlert prompt.
